### PR TITLE
Impl delay for withdrawals above a certain limit

### PR DESCRIPTION
**Pull request by zktony**
Impl delay for withdrawals above a certain limit.

### CODE REVIEW

**Comment by Gauthamastro** [comment]
@zktony An attacker can make multiple smaller withdrawals just below the limit and use a batch execution of them, to circumvent this guard, so having the delay for all withdrawals is better. Also, one might say, there is a front-running scenario there but I believe that is not the case for withdrawals to Ethereum and we should give priority to securing the user funds and protecting them with governance.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]
CHeck my comment above

File: pallets/asset-handler/src/lib.rs [hunk 1]
```diff
@@ -199,13 +288,18 @@ pub mod pallet {
...
 			);
+			ensure!(!<BridgeDeactivated<T>>::get(), Error::<T>::BridgeDeactivated);
 			let rid = chainbridge::derive_resource_id(chain_id, &contract_add.0);
 			ensure!(
 				T::AssetManager::reducible_balance(Self::convert_asset_id(rid), &sender, true) >=
 					amount.saturated_into::<u128>(),
 				Error::<T>::NotEnoughBalance
 			);
 			let fee = Self::fee_calculation(chain_id, amount);
+
```
Comment (Gauthamastro) [inline pallets/asset-handler/src/lib.rs]: Do this ensure before fee calculation

**Review by Gauthamastro — COMMENTED** [review]
- [x] Needs unit testing 
- [x] benchmarking

File: pallets/asset-handler/src/lib.rs (hunk 1 above)
Comment (zktony) [reply pallets/asset-handler/src/lib.rs]: done

File: pallets/asset-handler/src/lib.rs [hunk 2]
```diff
@@ -168,14 +168,14 @@ pub mod pallet {
...
-				pending_withdrawals.iter().map(|withdrawal| {
+				for withdrawal in pending_withdrawals {
 					chainbridge::Pallet::<T>::transfer_fungible(
 						withdrawal.chain_id,
 						withdrawal.rid,
 						withdrawal.recipient.0.to_vec(),
 						Self::convert_balance_to_eth_type(withdrawal.amount),
-					);
-				});
+					).expect("Chainbridge::transfer_fungible failed");
```
Comment (Gauthamastro) [inline pallets/asset-handler/src/lib.rs]: Can't have this.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/asset-handler/src/lib.rs (hunk 2 above)
Comment (zktony) [reply pallets/asset-handler/src/lib.rs]: removed

File: pallets/asset-handler/src/lib.rs [hunk 3]
```diff
@@ -174,7 +174,7 @@ pub mod pallet {
 						withdrawal.rid,
 						withdrawal.recipient.0.to_vec(),
 						Self::convert_balance_to_eth_type(withdrawal.amount),
-					).expect("Chainbridge::transfer_fungible failed");
+					);
```
Comment (Gauthamastro) [inline pallets/asset-handler/src/lib.rs]: Handle the result

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/asset-handler/src/lib.rs (hunk 3 above)
Comment (zktony) [reply pallets/asset-handler/src/lib.rs]: done

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**